src/main/java/jmgomez/policyapi/service/PolicyService.java

```diff
@@ -0,0 +1,48 @@
+package jmgomez.policyapi.service;
+
+import jmgomez.policyapi.feignClient.PolicyClient;
+import jmgomez.policyapi.mapper.AccidentMapper;
+import jmgomez.policyapi.mapper.PolicyMapper;
+import jmgomez.policyapi.model.dto.AccidentDto;
+import jmgomez.policyapi.model.dto.PolicyDto;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.annotation.CachePut;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class PolicyService implements IPolicyService {
+
+    @Autowired
+    private PolicyClient policyClient;
+
+    @Autowired
+    private PolicyMapper policyMapper;
+
+    @Autowired
+    private AccidentMapper accidentMapper;
+
+    @Override
+    public List<PolicyDto> getPolicies(String userId) {
+        return policyMapper.toListDto(policyClient.getPolicies(userId));
+    }
+
+    @Override
+    @Cacheable("policies")
```

Comment:
Well I tried it and I tested it in an easy way that was using a print in the terminal. The first time this line appeared because the controller called the service's method but the second time this information was in the cache so the controller didn't call the service. No, I didn't tested if cache works better in the service or the controller.

Reply:
You should have checked the same in the CachePut one... ;). And I wasn't asking you if you have tested if cache works better in the service. I'm talking about convenience. What do you think it would be better suited if we have a bigger API, maybe not with a 1 to 1 relationship between controller and service methods, maybe some orchestration/composition... What would be better in a more complex situation?

I'm also interested in how this @Cacheable works, how it works? I can't see an "if" sentence. It's described in the annotation and that if could be in spring code before calling our Controller methods. But... if we move the @Cacheable annotation to our Service#doSomething method and we just call service.doSomething from our controller... will it work? what will we see? I want you to think about this. I'm asking you about it. Hint: testing it and debugging could help you LOTS to understand how this is working.